openfl/component/aggregator/aggregator.py

```diff
@@ -1104,11 +1107,18 @@ def _compute_validation_related_task_metrics(self, task_name) -> dict:
                 if "validate_agg" in tags:
                     # Compare the accuracy of the model, potentially save it.
                     if self.best_model_score is None or self.best_model_score < agg_results:
-                        logger.info(
-                            f"Round {round_number}: saved the best model with score {agg_results:f}"
-                        )
                         self.best_model_score = agg_results
-                        self._save_model(round_number, self.best_state_path)
+                        if not self.assigner.is_task_group_evaluation():
+                            logger.info(
+                                f"Round {round_number}: saved the best model with score "
+                                "{agg_results:f}"
```

Comment:
Bug: this prints as `Round 9: saved the best model with score {agg_results:f}` due to unformatted f-string.

I'll fix this in a subsequent PR.